workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/SpatialOSConnectionHandlerBuilder.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Threading.Tasks;
+using Improbable.Worker.CInterop;
+
+namespace Improbable.Gdk.Core
+{
+    public class SpatialOSConnectionHandlerBuilder : IConnectionHandlerBuilder
+    {
+        public enum ThreadingMode
+        {
+            SingleThreaded,
+            MultiThreaded
+        }
+
+        private ReceptionistConfig? receptionistConfig;
+        private LocatorConfig? locatorConfig;
+        private AlphaLocatorConfig? alphaLocatorConfig;
```

Comment:
Wish we have C# unions :confused: 

Alternatively - perfect use case for https://github.com/mcintyre321/OneOf but sadly external dependencies :sob:

Reply:
We can make this a union if we really want to. I wasn't changing any of this code though, just rearranging it.

Reply:
Also I think we can get rid of alpha locator though.

Reply:
nope, not of the alpha locator :P we could get rid of the locator, if  you wanna reduce complexity.

Reply:
Discussed offline - can't get rid of the locator.